crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl

```diff
@@ -2053,6 +2054,7 @@
   and 	 \$0xf,$length
   je 	 .L_ret_${rndsuffix}
   jmp 	 .L_steal_cipher_next_${rndsuffix}
+  vzeroupper
```

Comment:
So would it matter if this `vzeroupper` is moved before `je`? Would it be less performant? If it doesn't matter, why not move it (here and in other places with a conditional jump)?

Reply:
Yes. We "blindly" added "vzeroupper" before return.

As mentioned earlier, some of these are not needed.. as the code never gets reached. Above is one such instance. Its a dead code (line 2057 and below never gets executed because of unconditional jump above).  There is another similar instance. 

  add 	         \$0x10,$output
  vmovdqa 	 %xmm1,%xmm8
  jmp 	         .L_done_${rndsuffix}
  vzeroupper  <<<
  ret

We plan to clean up the patch.